Frameworks/Foundation/NSArray.mm

```diff
@@ -666,10 +666,10 @@ - (NSUInteger)indexOfObject:(NSObject*)obj
                                             &comparator);
 
     if (index >= [self count] && (options & NSBinarySearchingInsertionIndex)) {
-        return index;
+        return [self count];
```

Comment:
Please replace all the calls to [self count] in this method with a cached local variable